erb2epp

```diff
@@ -1,77 +1,106 @@
-#!/usr/bin/ruby
+#!/usr/bin/env ruby
 
 require 'temple'
 require 'ripper'
 
 # Rewrite code from Ruby to EPP
-#
-# This comes down to replacing @var with $var
-def rewrite_code(code)
-  tokens = Ripper.lex(code)
+class Erb2epp
+  def initialize
+    @vars_found = []
+  end
 
-  # First pass: rewrite
-  tokens.each do |token|
-    # [[lineno, column], type, token, state]
-    _, type, value, _state = token
-    case type
-    when :on_ivar
-      token[2] = value.gsub(/@([a-z][A-Za-z0-9_]*)/, '$\1')
-      # TODO: Store variable name somewhere to build a list of inputs
-    when :on_op
-      case value
-      when '||'
-        token[2] = 'or'
-      when '&&'
-        token[2] = 'and'
-      end
-    when :on_kw
-      case value
-      when 'end'
-        token[2] = '}'
-      when 'if'
-        # TODO: append { somewhere
+  # Rewrite a line of code
```

Comment:
Technically it doesn't have to be a line. It's just a part of Ruby code within the ERB. I think it also handles something like:
```erb
<%-
if x
  a = 1
else
  a = 2
end
-%>
```

Most notably, it's not a full Ruby program so you can't use a parser, but a lexer can work well. So I'd write something like:

```suggestion
  # Rewrite a piece of Ruby code
  #
  # The ruby code blocks in ERB are not full programs so a parser can't be used. A lexer is used instead.
```